### ISSUE TITLE
Rates API is down

### ISSUE THREAD

**Issue by teran-mckinney** [AustEcon/bitsv#30]
bitcoinsv-rates.com is no longer resolving. Not having any luck getting rates with bitsv at the moment.

Does anyone know of the owners of that site? Are there any other rates APIs we can use?

**Comment by teran-mckinney** [comment]
Ok, so the DNS host is not replying with any data, the webserver is up. It's at 195.154.226.214.

**Comment by teran-mckinney** [comment]
I see no contact information, I'll message on Atlantis.

**Comment by AustEcon** [comment]
Thanks dude!

**Comment by teran-mckinney** [comment]
No problem! Unfortunately, looks the Whois data is privacy protected but doesn't have a unique email, so not sure how to contact them. No email address on the website from what I can tell.

**Comment by AustEcon** [comment]
I've direct messaged that username from reddit...
I'm already planning for worst case scenario...
We have this: 
https://api.bitfinex.com/v1/pubticker/bsvusd
Which combined with this: 
https://www.freeforexapi.com/api/live?pairs=NZDUSD 

We can build a replacement if needed...

**Comment by teran-mckinney** [comment]
Nice! We should add some redundancy in, for sure. Hopefully this will be fixed soon.

**Comment by AustEcon** [comment]
Hmmm. okay @teran-mckinney and @joshua-s - It's not looking promising... Think these endpoints will be okay? I'm pretty time poor right now... I can have a play around and see what I can get going but otherwise let me know if either of you are putting in any efforts on this front, I will gladly step aside 😉 !

**Comment by teran-mckinney** [comment]
Agreed, I'd call it pretty hard down. I wonder if someone will make an on-blockchain rates API?

Anyway though, I'll see if I can dig into this.

**Comment by AustEcon** [comment]
Cool. There is already a caching system there so I think it won't be too bad if we use the bitfinex usd rate to derive the other ones... because of 60 second cache... I find wrapper functions really confusing though to work with!

**Comment by AustEcon** [comment]
I've got it working... only rudimentary testing done at this stage... Maybe you will have a better implementation though... I'll put up what I've got.

**Comment by teran-mckinney** [comment]
Ok, will review tomorrow. Thank you!

**Comment by teran-mckinney** [comment]
You won't guess what just started working!

**Comment by AustEcon** [comment]
Lol! Okay. Well that's good though. Maybe we can add this all as a second option as redundancy so it doesn't happen again!

Should be fairly trivial. The framework is already there to do it... can just duplicate and add the two classes in series...

**Comment by teran-mckinney** [comment]
Yup, just like the transactions APIs, just fail from one to another.

**Comment by teran-mckinney** [comment]
Nice work, thank you! Let me look into bitcoinsv-rates.com, it seems to work from some places and not others.

**Comment by teran-mckinney** [comment]
Ok, maybe it came up for a bit, got some results cached, and is down again. Weird.

```
root@gobeyond-1:~# drill -T bitcoinsv-rates.com
.       518400  IN      NS      l.root-servers.net.
.       518400  IN      NS      d.root-servers.net.
.       518400  IN      NS      e.root-servers.net.
.       518400  IN      NS      k.root-servers.net.
.       518400  IN      NS      c.root-servers.net.
.       518400  IN      NS      h.root-servers.net.
.       518400  IN      NS      b.root-servers.net.
.       518400  IN      NS      a.root-servers.net.
.       518400  IN      NS      i.root-servers.net.
.       518400  IN      NS      f.root-servers.net.
.       518400  IN      NS      j.root-servers.net.
.       518400  IN      NS      m.root-servers.net.
.       518400  IN      NS      g.root-servers.net.
com.    172800  IN      NS      m.gtld-servers.net.
com.    172800  IN      NS      k.gtld-servers.net.
com.    172800  IN      NS      i.gtld-servers.net.
com.    172800  IN      NS      e.gtld-servers.net.
com.    172800  IN      NS      h.gtld-servers.net.
com.    172800  IN      NS      g.gtld-servers.net.
com.    172800  IN      NS      a.gtld-servers.net.
com.    172800  IN      NS      j.gtld-servers.net.
com.    172800  IN      NS      b.gtld-servers.net.
com.    172800  IN      NS      c.gtld-servers.net.
com.    172800  IN      NS      l.gtld-servers.net.
com.    172800  IN      NS      f.gtld-servers.net.
com.    172800  IN      NS      d.gtld-servers.net.
bitcoinsv-rates.com.    172800  IN      NS      dns1.turhost.com.
bitcoinsv-rates.com.    172800  IN      NS      dns2.turhost.com.
root@gobeyond-1:~# drill bitcoinsv-rates.com @dns1.turhost.com
;; ->>HEADER<<- opcode: QUERY, rcode: REFUSED, id: 261
;; flags: qr rd ; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; bitcoinsv-rates.com. IN      A

;; ANSWER SECTION:

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 138 msec
;; SERVER: 185.15.40.93
;; WHEN: Mon Jul  8 15:03:09 2019
;; MSG SIZE  rcvd: 37
root@gobeyond-1:~# drill bitcoinsv-rates.com @dns2.turhost.com
;; ->>HEADER<<- opcode: QUERY, rcode: REFUSED, id: 44572
;; flags: qr rd ; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; bitcoinsv-rates.com. IN      A

;; ANSWER SECTION:

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 136 msec
;; SERVER: 185.15.40.94
;; WHEN: Mon Jul  8 15:03:14 2019
;; MSG SIZE  rcvd: 37
```

**Comment by joshua-s** [comment]
I believe this is now resolved in 0.8.0? If so, let's close it

**Comment by teran-mckinney** [comment]
Yes, and bitcoinsv-rates.com is back up O_o. Finally!

Thank you!
